includes/class-sensei-admin.php

```diff
@@ -1418,11 +1418,13 @@ public function lesson_order_screen() {
 			$html .= '<option value="">' . esc_html__( 'Select a course', 'sensei-lms' ) . '</option>' . "\n";
 
 			foreach ( $courses as $course ) {
-				$course_id = '';
-				if ( isset( $_GET['course_id'] ) ) {
-					$course_id = intval( $_GET['course_id'] );
+				if ( current_user_can( 'edit_others_posts' ) || get_current_user_id() === (int) $course->post_author ) {
```

Comment:
I was thinking that instead of checking inside the loop, we can modify the query. That way if the site has a gazillion ([it's a word!](https://www.merriam-webster.com/dictionary/gazillion) 😄) courses, we won't get them all. Something like this might do the trick:

```php
// Make sure the user has permissions to edit the courses of others.
if ( ! current_user_can( 'edit_others_courses' ) ) {
		$args['author'] = get_current_user_id();
}
```

Reply:
Thanks @m1r0 - this makes a lot more sense.  Good catch!